pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -3157,42 +3157,38 @@ public CompletableFuture<Boolean> isAllowAutoTopicCreationAsync(final String top
         TopicName topicName = TopicName.get(topic);
         return isAllowAutoTopicCreationAsync(topicName);
     }
-
     public CompletableFuture<Boolean> isAllowAutoTopicCreationAsync(final TopicName topicName) {
-        Optional<Policies> policies =
-                pulsar.getPulsarResources().getNamespaceResources()
-                        .getPoliciesIfCached(topicName.getNamespaceObject());
-        return isAllowAutoTopicCreationAsync(topicName, policies);
-    }
-
-    private CompletableFuture<Boolean> isAllowAutoTopicCreationAsync(final TopicName topicName,
-                                                                     final Optional<Policies> policies) {
-        if (policies.isPresent() && policies.get().deleted) {
-            log.info("Preventing AutoTopicCreation on a namespace that is being deleted {}",
-                    topicName.getNamespaceObject());
-            return CompletableFuture.completedFuture(false);
-        }
-        //System topic can always be created automatically
-        if (pulsar.getConfiguration().isSystemTopicEnabled() && isSystemTopic(topicName)) {
-            return CompletableFuture.completedFuture(true);
-        }
-        final boolean allowed;
-        AutoTopicCreationOverride autoTopicCreationOverride = getAutoTopicCreationOverride(topicName, policies);
-        if (autoTopicCreationOverride != null) {
-            allowed = autoTopicCreationOverride.isAllowAutoTopicCreation();
-        } else {
-            allowed = pulsar.getConfiguration().isAllowAutoTopicCreation();
-        }
-
-        if (allowed && topicName.isPartitioned()) {
-            // cannot re-create topic while it is being deleted
-            return pulsar.getPulsarResources().getNamespaceResources().getPartitionedTopicResources()
-                    .isPartitionedTopicBeingDeletedAsync(topicName)
-                    .thenApply(beingDeleted -> !beingDeleted);
-        } else {
-            return CompletableFuture.completedFuture(allowed);
-        }
-
+        return pulsar.getPulsarResources().getNamespaceResources()
+                .getPoliciesAsync(topicName.getNamespaceObject(), true)
```

Comment:
I think that there maybe some case in which it is good to use the local version of the Policies object.

We need to use "refresh" only when we reach here from a HTTP API call,
I know that is looks pretty convoluted, but with this change in the current form we are going to use "refresh"  (that is a heavyweight operation) probably in code paths that don't need it

Reply:
+1

Reply:
I traced http vs binary protocol calls that invoke this code:
http calls:
-  resetCursorOnPosition
-  lookupTopic
-  createSubscription
-  grantPermissionsOnTopic
-  getPartitionedMetadata
-  Basically any method calling the [AdminResource#getPartitionedTopicMetadataAsync](https://github.com/apache/pulsar/blob/416c5b40ac1c44bbe281dba35ea6682dae7697da/pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java#L482) with `checkAllowAutoCreation` true

binary protocol calls:
- handlePartitionMetadataRequest
- handleSubscribe
- handleProducer 

I isolated the calls with as minimum code changes as possible. Please evaluate the value of the change vs the introduced code branches. 

PTAL.